src/components/ArticlesList.jsx

```diff
@@ -1,54 +1,71 @@
 import { useEffect, useState } from "react"
+import { useSearchParams } from "react-router-dom"
+import { getArticles } from "../utils/api"
 import ArticleCard from "./ArticleCard"
-import { getArticles, getArticlesByTopic } from "../utils/api"
-import { useParams } from "react-router-dom"
+import SortOptions from "./SortOptions"
+import Navbar from "./Navbar"
 
 const ArticlesList = () => {
 
-    const { topic } = useParams()
+    const [searchParams, setSearchParams] = useSearchParams()
+    const topicQuery = searchParams.get("topic")
+    const sortByQuery = searchParams.get("sort_by")
+    const orderQuery = searchParams.get("order")
+
     const [articles, setArticles] = useState([])
     const [isLoading, setIsLoading] = useState(true)
     const [isError, setIsError] = useState(false)
 
+    const setTopic = (topic) => {
+        const newParams = new URLSearchParams(searchParams)
+        newParams.set('topic', topic)
+        setSearchParams(newParams)
+    }
+
+    const setSortBy = (sort_by) => {
+        const newParams = new URLSearchParams(searchParams)
+        newParams.set('sort_by', sort_by)
+        setSearchParams(newParams)
+    }
+
+    const setOrder = (order) => {
+        const newParams = new URLSearchParams(searchParams)
+        newParams.set('order', order)
+        setSearchParams(newParams)
+    }
+
```

Comment:
Some repeating code here, you might want to consider how you could refactor this into one function to streamline the code a bit 🚄